### PR TITLE
Added Nullcheck in getting JobStreamRecord values and added Runbook, Job, JobOutput tests

### PR DESCRIPTION
**Pull request by NarayanThiru**
Added Runbook, Job and JobOutput Tests and test artifacts

## Description

Added Nullcheck in getting JobStreamRecord values
Added Runbook, Job, JobOutput tests

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by MiYanni** [comment]
@NarayanThiru You have 3 test failures:

```powershell
Commands.Automation.Test.RunbookJobTests.TestCreateRunbookGraph
System.ArgumentException : Unable to find recorded mock file 'c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\bin\Debug\SessionRecords\Commands.Automation.Test.RunbookJobTests\TestCreateRunbookGraph.json'.\r\nParameter name: callerIdentity
   at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode)
   at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName)
   at Microsoft.Azure.Commands.Automation.Test.AutomationScenarioTestsBase.RunPowerShellTest(XunitTracingInterceptor logger, String[] scripts) in c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\ScenarioTests\AutomationScenarioTestsBase.cs:line 50
   at Commands.Automation.Test.RunbookJobTests.TestCreateRunbookGraph() in c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\ScenarioTests\RunbookJobTests.cs:line 40
Traits:
Service	Automation
RunType	LiveOnly
AcceptanceType	CheckIn
AcceptanceType	BVT
Commands.Automation.Test.RunbookJobTests.TestImportAndDeleteRunbookGraphical
System.ArgumentException : Unable to find recorded mock file 'c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\bin\Debug\SessionRecords\Commands.Automation.Test.RunbookJobTests\TestImportAndDeleteRunbookGraphical.json'.\r\nParameter name: callerIdentity
   at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode)
   at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName)
   at Microsoft.Azure.Commands.Automation.Test.AutomationScenarioTestsBase.RunPowerShellTest(XunitTracingInterceptor logger, String[] scripts) in c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\ScenarioTests\AutomationScenarioTestsBase.cs:line 50
   at Commands.Automation.Test.RunbookJobTests.TestImportAndDeleteRunbookGraphical() in c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\ScenarioTests\RunbookJobTests.cs:line 60
Traits:
Service	Automation
RunType	LiveOnly
AcceptanceType	CheckIn
AcceptanceType	BVT
Commands.Automation.Test.RunbookJobTests.TestImportRunbookPowerShell
System.ArgumentException : Unable to find recorded mock file 'c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\bin\Debug\SessionRecords\Commands.Automation.Test.RunbookJobTests\TestImportRunbookPowerShell.json'.\r\nParameter name: callerIdentity
   at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode)
   at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName)
   at Microsoft.Azure.Commands.Automation.Test.AutomationScenarioTestsBase.RunPowerShellTest(XunitTracingInterceptor logger, String[] scripts) in c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\ScenarioTests\AutomationScenarioTestsBase.cs:line 50
   at Commands.Automation.Test.RunbookJobTests.TestImportRunbookPowerShell() in c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\ScenarioTests\RunbookJobTests.cs:line 50
Traits:
Service	Automation
RunType	LiveOnly
AcceptanceType	CheckIn
AcceptanceType	BVT
```

I noticed you have the marked as both `LiveOnly` and as `CheckIn`. We cannot do live tests at check-in. If they are intended to be `LiveOnly`, please remove the `CheckIn` trait.

File: src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/RunbookJobTests.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            // Write PS Function and call it here
+            RunPowerShellTest(_logger, "Test-CreateRunbookGraph -Name RB-Graphical-01");
+        }
+
+        [Fact]
+        [Trait(Category.Service, Category.Automation)]
+        [Trait(Category.RunType, Category.LiveOnly)]
+        public void TestImportRunbookPowerShell()
+        {
+            RunPowerShellTest(_logger, "Test-ImportRunbookPowerShell -Name TestRunbook-PowerShellScript -RunbookPath ScenarioTests\\Resources\\RB-PowerShellScriptTutorial.ps1");
```
Comment (MiYanni) [inline src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/RunbookJobTests.cs]: The `RunbookPath` provided here will not work on Linux. Please use something like:
```cs
$"Test-ImportRunbookPowerShell -Name TestRunbook-PowerShellScript -RunbookPath {Path.Combine("ScenarioTests", "Resources", "RB-PowerShellScriptTutorial.ps1")}"
```

File: src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/RunbookJobTests.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        [Fact]
+        [Trait(Category.Service, Category.Automation)]
+        [Trait(Category.RunType, Category.LiveOnly)]
+        public void TestImportAndDeleteRunbookGraphical()
+        {
+            RunPowerShellTest(_logger, "Test-ImportAndDeleteRunbookGraphical -Name TestRunbook-Grapical -RunbookPath ScenarioTests\\Resources\\RB-GraphTutorial.graphrunbook");
```
Comment (MiYanni) [inline src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/RunbookJobTests.cs]: See above comment about `RunbookPath`.

File: src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/RunbookJobTests.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        [Fact(Skip = "Failed output record causes JSON convert exception in Travis run.")]
+        [Trait(Category.Service, Category.Automation)]
+        [Trait(Category.RunType, Category.LiveOnly)]
+        public void TestCreateJobAndGetOutputPowerShellScript()
+        {
+            RunPowerShellTest(_logger, "Test-CreateJobAndGetOutputPowerShellScript -Name TestRunbook-PSScript-JobAndOutput -RunbookPath ScenarioTests\\Resources\\RB-PowerShellScriptTutorial.ps1");
```
Comment (MiYanni) [inline src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/RunbookJobTests.cs]: See above comment about `RunbookPath`.

File: src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/RunbookJobTests.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+        public RunbookJobTests(Xunit.Abstractions.ITestOutputHelper output)
+        {
+            _logger = new XunitTracingInterceptor(output);
+            XunitTracingInterceptor.AddToContext(_logger);
+        }
+
+        [Fact]
+        [Trait(Category.Service, Category.Automation)]
+        [Trait(Category.RunType, Category.LiveOnly)]
```
Comment (MiYanni) [inline src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/RunbookJobTests.cs]: Is there a reason that every test being added is `LiveOnly`?

**Comment by MiYanni** [comment]
@NarayanThiru You have test failures:
```powershell
Commands.Automation.Test.RunbookJobTests.TestCreateRunbookGraph
System.ArgumentException : Unable to find recorded mock file 'c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\bin\Debug\SessionRecords\Commands.Automation.Test.RunbookJobTests\TestCreateRunbookGraph.json'.\r\nParameter name: callerIdentity
   at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode)
   at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName)
   at Microsoft.Azure.Commands.Automation.Test.AutomationScenarioTestsBase.RunPowerShellTest(XunitTracingInterceptor logger, String[] scripts) in c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\ScenarioTests\AutomationScenarioTestsBase.cs:line 54
   at Commands.Automation.Test.RunbookJobTests.TestCreateRunbookGraph() in c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\ScenarioTests\RunbookJobTests.cs:line 39
Traits:
Service	Automation
AcceptanceType	CheckIn

Commands.Automation.Test.RunbookJobTests.TestImportAndDeleteRunbookGraphical
System.ArgumentException : Unable to find recorded mock file 'c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\bin\Debug\SessionRecords\Commands.Automation.Test.RunbookJobTests\TestImportAndDeleteRunbookGraphical.json'.\r\nParameter name: callerIdentity
   at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode)
   at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName)
   at Microsoft.Azure.Commands.Automation.Test.AutomationScenarioTestsBase.RunPowerShellTest(XunitTracingInterceptor logger, String[] scripts) in c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\ScenarioTests\AutomationScenarioTestsBase.cs:line 54
   at Commands.Automation.Test.RunbookJobTests.TestImportAndDeleteRunbookGraphical() in c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\ScenarioTests\RunbookJobTests.cs:line 57
Traits:
Service	Automation
AcceptanceType	CheckIn

Commands.Automation.Test.RunbookJobTests.TestImportRunbookPowerShell
System.ArgumentException : Unable to find recorded mock file 'c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\bin\Debug\SessionRecords\Commands.Automation.Test.RunbookJobTests\TestImportRunbookPowerShell.json'.\r\nParameter name: callerIdentity
   at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode)
   at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName)
   at Microsoft.Azure.Commands.Automation.Test.AutomationScenarioTestsBase.RunPowerShellTest(XunitTracingInterceptor logger, String[] scripts) in c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\ScenarioTests\AutomationScenarioTestsBase.cs:line 54
   at Commands.Automation.Test.RunbookJobTests.TestImportRunbookPowerShell() in c:\workspace\powershell\src\ResourceManager\Automation\Commands.Automation.Test\ScenarioTests\RunbookJobTests.cs:line 48
Traits:
Service	Automation
AcceptanceType	CheckIn
```

**Pull request merged**